### PR TITLE
feat(PLP): Category Page Product Listings.

### PR DESCRIPTION
**Pull request by kibo-geetanshuchhabra**
@kibo-kevinwatts @kibo-sushant

### CODE REVIEW

File: pages/category/index.vue [hunk 1]
```diff
@@ -73,13 +73,91 @@
...
+          <transition-group
+            v-if="isGridView"
+            appear
+            name="products__slide"
+            tag="div"
+            class="products__grid"
+          >
+            <KiboProductCard
+              v-for="(product, i) in products"
+              :key="productGetters.getSlug(product)"
```
Comment (kibo-kevinwatts) [inline pages/category/index.vue]: use product code for key, not all products will have slug

File: pages/category/index.vue [hunk 2]
```diff
@@ -73,13 +73,91 @@
...
+              :maxRating="5"
+              wishlistIcon=""
+              isInWishlistIcon=""
+              :isInWishlist="false"
+              :show-add-to-cart-button="true"
+              :regularPrice="`$${productGetters.getPrice(product).regular}`"
+              :special-price="
+                productGetters.getPrice(product).special && productGetters.getPrice(product).special
+              "
+              :link="localePath(`/product/${productGetters.getProductId(product)}`)"
```
Comment (kibo-kevinwatts) [inline pages/category/index.vue]: create helper for getting product url, similar to category link helper

File: pages/category/index.vue [hunk 3]
```diff
@@ -73,13 +73,91 @@
...
+              imageWidth="12.563rem"
+              imageHeight="12.563rem"
+              class="products__product-card"
+            />
+          </transition-group>
+          <transition-group v-else appear name="products__slide" tag="div" class="products__list">
+            <SfProductCardHorizontal
+              v-for="(product, i) in products"
+              :key="productGetters.getSlug(product)"
```
Comment (kibo-kevinwatts) [inline pages/category/index.vue]: use product code for key

File: pages/category/index.vue [hunk 4]
```diff
@@ -73,13 +73,91 @@
...
+                />
+                <SfProperty class="desktop-only" name="Color" value="white" />
+              </template>
+              <template #actions>
+                <SfButton
+                  class="sf-button--text desktop-only"
+                  style="margin: 0 0 1rem auto; display: block"
+                  @click="$emit('click:add-to-wishlist')"
+                >
+                  Save for later
```
Comment (kibo-kevinwatts) [inline pages/category/index.vue]: use i18n to get string

File: pages/category/index.vue [hunk 5]
```diff
@@ -73,13 +73,91 @@
...
+                </SfButton>
+                <SfButton
+                  class="sf-button--text desktop-only"
+                  style="margin: 0 0 0 auto; display: block"
+                  @click="$emit('click:add-to-compare')"
+                >
+                  Add to compare
```
Comment (kibo-kevinwatts) [inline pages/category/index.vue]: use i18n to get string

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: pages/category/index.vue (hunk 1 above)
Comment (kibo-geetanshuchhabra) [reply pages/category/index.vue]: Resolved

File: pages/category/index.vue (hunk 4 above)
Comment (kibo-geetanshuchhabra) [reply pages/category/index.vue]: Resolved

File: pages/category/index.vue (hunk 5 above)
Comment (kibo-geetanshuchhabra) [reply pages/category/index.vue]: Resolved

File: pages/category/index.vue [hunk 6]
```diff
@@ -115,16 +194,27 @@ export default {
...
-    const { getFacetsFromURL, getCatLink } = useUiHelpers()
+    const { getFacetsFromURL, getCatLink, getProductLink } = useUiHelpers()
     const { result, search, loading } = useFacet(`category-listing`)
+    const {
+      result: productSearchResult,
+      search: productSearch,
+      loading: productSearchLoading,
+    } = useProductSearch(`product-search`)
+
+    const loadingCategory = useState("category-loading", () => false)
```
Comment (kibo-kevinwatts) [inline pages/category/index.vue:209]: composable should handle toggling of load state

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**